pkg/status/status.go

```diff
@@ -44,23 +37,19 @@ const (
 	ConditionTypeIncorrectNUMAResourcesOperatorResourceName = "IncorrectNUMAResourcesOperatorResourceName"
 )
 
-func Update(ctx context.Context, client k8sclient.Client, rte *nropv1alpha1.NUMAResourcesOperator, condition string, reason string, message string) (bool, error) {
+func UpdateConditions(currentConditions []metav1.Condition, condition string, reason string, message string) ([]metav1.Condition, bool) {
```

Comment:
this is nice, but doesn't really update conditions, does it? A name like `GetUpdatedConditions` seems a bit more fitting. Besides the name, the rest seems just fine.